File: ruby/Flyweight/flight.rb
class Flight
  attr_accessor :company

  def initialize (company)
    @company = company
  end

  def draw_on_map (flight_no,latitude, longitude)
    puts "#{@company}  flight  #{flight_no}  is currently at #{latitude}  latitude  and #{longitude}  longitude"
  end
end

File: java/Flyweight/src/flyweight/Flight.java
package flyweight;

public class Flight {
	String company;
	
	void drawOnMap(int flightNo, double latitude, double longitude) {
		System.out.println();
		System.out.println(company + " flight " + flightNo + " is currently at latitude " + latitude + " and longitude "
				+ longitude);
	}
}
